Addressing issue #38325

low-perry opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

Updated the definition of the `returnValue` property, to better align with it's function.
Updated the example of `returnValue` usage for better clarity

### Motivation

I made these changes to help readers better understand the `returnValue` property of the `<dialog>` element, and provided an example that demonstrates in a clearer way how this property is used.


<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

"Fixes #38325"
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

low-perry committed:
Update definition of `returnValue` of the <dialog> element

low-perry committed:
Change example for clarity

low-perry committed:
Fix inline html element `<kbd>`

low-perry committed:
Format Example paragraph. Fix HTML identation.

low-perry committed:
Update files/en-us/web/api/htmldialogelement/returnvalue/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

low-perry committed:
Update files/en-us/web/api/htmldialogelement/returnvalue/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

low-perry committed:
Fix identation in HTML/JS example

bsmth committed:
Merge branch 'main' into patch-8

bsmth commented on files/en-us/web/api/htmldialogelement/returnvalue/index.md:
```diff
@@ -8,102 +8,77 @@ browser-compat: api.HTMLDialogElement.returnValue
+  const declineButton = document.getElementById("declineButton");
+  const acceptButton = document.getElementById("acceptButton");
+
+  function openCheck(dialog) {
```
Let's remove the openCheck so we can focus on what happens when closed only.

low-perry replied:
Will do. Thank you.

bsmth commented on files/en-us/web/api/htmldialogelement/returnvalue/index.md:
```diff
@@ -8,102 +8,77 @@ browser-compat: api.HTMLDialogElement.returnValue
+    dialog.close("accepted");
+    openCheck(dialog);
+    handleUserInput(dialog.returnValue);
+  });
```
There's a few ways to do this, we could also do this if you like it:

```js
declineButton.addEventListener("click", closeDialog);
acceptButton.addEventListener("click", closeDialog);

function closeDialog(event) {
  const button = event.target;
  const returnValue = button.value;
  dialog.close(returnValue);
}
```

bsmth replied:
The nice thing about yours is we show passing a string explicitly as `.close("accepted")`

low-perry replied:
I think your code is better, it removes the unnecessary duplication, and it's pretty clear. I can add this change.

bsmth commented on files/en-us/web/api/htmldialogelement/returnvalue/index.md:
```diff
@@ -8,102 +8,77 @@ browser-compat: api.HTMLDialogElement.returnValue
 {{ APIRef("HTML DOM") }}
 
-The **`returnValue`** property of the {{domxref("HTMLDialogElement")}} interface gets or sets the return value for the {{htmlelement("dialog")}}, usually to indicate which button the user pressed to close it.
+The **`returnValue`** property of the {{domxref("HTMLDialogElement")}} interface is a string that stores the value provided when the {{htmlelement("dialog")}} is closed using the `.close(returnValue)` method.
```
The original is correct here, you can get or set the return value. Also you can close the dialog (when using `showModal`) with:

* `<button formmethod="dialog">Close</button>`
* ESC key
* light dismiss (clicking outside the dialog)

low-perry replied:
I totally agree with that, you can set or get the property, but my concern is that the original phrasing says that the property performs an action (gets or sets). Maybe it's just me being pedantic or I don't understand interface properties correctly.

bsmth replied:
No, you're right, if we check the IDL https://html.spec.whatwg.org/multipage/interactive-elements.html#the-dialog-element:dom-dialog-returnvalue is what you get with `close()` and `requestClose()`, so we can improve on it:

```suggestion
The **`returnValue`** property of the {{domxref("HTMLDialogElement")}} interface is a string representing the return value for a {{htmlelement("dialog")}} element when it's closed.
You can set the value directly (`dialog.returnValue = "result"`) or by providing the value as a string argument to `dialog.close()` or `dialog.requestClose()`.
```

bsmth commented:
Thanks a lot @low-perry - couple of small suggestions to apply, but looking good!

low-perry committed:
Remove unnecessary method , Combine the functions that close the dialog into a single one.

These changes are based on the comments left on my proposed changes:
1. [Remove `openCheck` method](https://github.com/mdn/content/pull/38359/files/cdea48036687a54cde93f5b6d119a3a7e252cc1b#r1975216037)
2. [Combine the functions](https://github.com/mdn/content/pull/38359/files/cdea48036687a54cde93f5b6d119a3a7e252cc1b#r1975219487)

bsmth commented on files/en-us/web/api/htmldialogelement/returnvalue/index.md:
```diff
@@ -8,102 +8,66 @@ browser-compat: api.HTMLDialogElement.returnValue
+  const acceptButton = document.getElementById("acceptButton");
+
+  function handleUserInput(returnValue) {
+    if (!returnValue) {
```
```suggestion
    if (returnValue !== "") {
```

we should check for equality to `""` empty string, because `0` as a return value would satisfy that

low-perry replied:
I agree 😊

bsmth commented on files/en-us/web/api/htmldialogelement/returnvalue/index.md:
```diff
@@ -8,102 +8,66 @@ browser-compat: api.HTMLDialogElement.returnValue
+The following example displays a button to open a dialog containing a Terms of Service prompt via the `showModal()` method.
+The script checks if the dialog is open or closed and updates the status text accordingly. It also handles the user's input by assigning the `returnValue` when the Accept or Decline button is clicked.
+The "Decline" button sets the `returnValue` to "declined", while the "Accept" button sets it to "accepted". Additionally, closing the dialog (e.g., using the close event) updates the status text with the dialog's `returnValue`.
+Closing the dialog with the <kbd>Esc</kbd> key has no impact on the returnValue.
```
```suggestion
Closing the dialog with the <kbd>Esc</kbd> key does not set the `returnValue`.
```

bsmth commented on files/en-us/web/api/htmldialogelement/returnvalue/index.md:
```diff
@@ -8,102 +8,66 @@ browser-compat: api.HTMLDialogElement.returnValue
-The confirm button (`confirmBtn`) submits the form with validation and the "X" button submits the form without validation. Submitting a form with a `method="dialog"` closes the dialog and sets the return value to the `value`, if any, of the `button` or `input` elements of `type=submit`.
-The reset button has an event handler that closes the dialog; it has no impact on the `returnValue`. Neither does closing the dialog with the <kbd>Esc</kbd> key.
+The following example displays a button to open a dialog containing a Terms of Service prompt via the `showModal()` method.
+The script checks if the dialog is open or closed and updates the status text accordingly. It also handles the user's input by assigning the `returnValue` when the Accept or Decline button is clicked.
```
```suggestion
The script handles the user's input by assigning the `returnValue` when the Accept or Decline button is clicked.
```

low-perry committed:
Update files/en-us/web/api/htmldialogelement/returnvalue/index.md

Co-authored-by: Brian Smith <brian@smith.berlin>

low-perry committed:
Update files/en-us/web/api/htmldialogelement/returnvalue/index.md

Co-authored-by: Brian Smith <brian@smith.berlin>

bsmth commented on files/en-us/web/api/htmldialogelement/returnvalue/index.md:
```diff
@@ -8,102 +8,66 @@ browser-compat: api.HTMLDialogElement.returnValue
 
 ## Value
 
 A string representing the `returnValue` of the dialog.
```
```suggestion
A string representing the `returnValue` of the dialog.
Defaults to an empty string (`""`).
```

low-perry committed:
Update files/en-us/web/api/htmldialogelement/returnvalue/index.md

Co-authored-by: Brian Smith <brian@smith.berlin>

low-perry committed:
Update files/en-us/web/api/htmldialogelement/returnvalue/index.md

Co-authored-by: Brian Smith <brian@smith.berlin>

low-perry committed:
Merge branch 'main' into patch-8

low-perry commented:
@bsmth I committed your changes. I think it looks way better now! 😊

bsmth commented:
> @bsmth I committed your changes. I think it looks way better now! 😊

That's great, looking really good, there's one comment that we should get in (mentioning `close()` and `requestClose()` in the intro) https://github.com/mdn/content/pull/38359#discussion_r1975537147 and then I think we're good to go

low-perry committed:
Update files/en-us/web/api/htmldialogelement/returnvalue/index.md

Co-authored-by: Brian Smith <brian@smith.berlin>

low-perry commented:
Sorry, I missed that. It should be good now!

bsmth committed:
Merge branch 'main' into patch-8

bsmth commented on files/en-us/web/api/htmldialogelement/returnvalue/index.md:
```diff
@@ -8,102 +8,68 @@ browser-compat: api.HTMLDialogElement.returnValue
 
-The **`returnValue`** property of the {{domxref("HTMLDialogElement")}} interface gets or sets the return value for the {{htmlelement("dialog")}}, usually to indicate which button the user pressed to close it.
+The **`returnValue`** property of the {{domxref("HTMLDialogElement")}} interface is a string representing the return value for a {{htmlelement("dialog")}} element when it's closed.
+You can set the value directly (`dialog.returnValue = "result"`) or by providing the value as a string argument to `dialog.close()` or `dialog.requestClose()`.
```
Just going to link for convenience:

```suggestion
You can set the value directly (`dialog.returnValue = "result"`) or by providing the value as a string argument to {{domxref("HTMLDialogElement.close()", "close()")}} or {{domxref("HTMLDialogElement.requestClose()", "requestClose()")}}.
```

bsmth reviewed (approved):
Thanks a lot! Looking much better, going to merge shortly

bsmth committed:
Update files/en-us/web/api/htmldialogelement/returnvalue/index.md
